eigen/client.go

```diff
@@ -98,66 +75,98 @@ func (e *Client) GetProof(ctx context.Context, id da.ID) (da.Proof, error) {
 	return nil, nil
 }
 
-type ID struct {
-	BlobIndex       uint32
-	BatchHeaderHash []byte
+// PutBlob encodes and writes a blob to EigenDA, waiting for it to be confirmed
+// before returning. This function is resiliant to transient failures and
+// timeouts.
+func (c *Client) PutBlob(ctx context.Context, data []byte) (*grpcdisperser.BlobInfo, error) {
+	resultChan, errorChan := c.PutBlobAsync(ctx, data)
+	select { // no timeout here because we depend on the configured timeout in PutBlobAsync
+	case result := <-resultChan:
+		return result, nil
+	case err := <-errorChan:
+		return nil, err
+	}
 }
 
-func (e *Client) disperseBlob(ctx context.Context, txData []byte) (*BlobInfo, error) {
-	fmt.Println("Attempting to disperse blob to EigenDA")
+func (c *Client) PutBlobAsync(ctx context.Context, data []byte) (resultChan chan *grpcdisperser.BlobInfo, errChan chan error) {
+	resultChan = make(chan *grpcdisperser.BlobInfo, 1)
+	errChan = make(chan error, 1)
+	go c.putBlob(ctx, data, resultChan, errChan)
+	return
+}
 
-	disperseReq := &DisperseBlobRequest{
-		Data:           txData,
-		SecurityParams: e.DADisperserSecurityParams,
+func (c *Client) putBlob(ctx context.Context, rawData []byte, resultChan chan *grpcdisperser.BlobInfo, errChan chan error) {
+	// encode blob
+	if c.internalClient.Codec == nil {
+		errChan <- fmt.Errorf("codec cannot be nil")
+		return
 	}
-	daClient := e.disperserClient
-	disperseRes, err := daClient.DisperseBlob(ctx, disperseReq)
-	fmt.Println("DisperseBlob response", "disperseRes", disperseRes, "err", err)
 
+	data, err := c.internalClient.Codec.EncodeBlob(rawData)
 	if err != nil {
-		fmt.Printf("Unable to disperse blob to EigenDA, aborting", "err", err)
-		return nil, err
+		errChan <- fmt.Errorf("error encoding blob: %w", err)
+		return
 	}
 
-	if disperseRes.Result == BlobStatus_UNKNOWN ||
-		disperseRes.Result == BlobStatus_FAILED {
-		fmt.Printf("Unable to disperse blob to EigenDA, aborting", "err", err)
-		return nil, fmt.Errorf("reply status is %d", disperseRes.Result)
+	customQuorumNumbers := make([]uint8, len(c.internalClient.Config.CustomQuorumIDs))
```

Comment:
Should these config variables like `CustomQuorumIDs`, `StatusQueryRetryInterval` and `StatusQueryTimeout` be made configurable?

Reply:
Yes they can be configured when creating a new `Client` here:
https://github.com/stackrlabs/go-daash/pull/19/files/901ae265bdf4eb7d0fd39bc0aa027b9e7cc4bf1a#diff-8afc4e06ff4735a8c36127188b82f42c2424a4272cfee63570860454a78299c0R25-R27